### PR TITLE
Added endpoints to send company notification email

### PR DESCRIPTION
**Pull request by jfkonecn**
closes oncokb/oncokb-pipeline#444

@zhx828 

Right now I'm doing an update for the entire company table. I'd prefer to just update the additional info column. What's the best way to do that?

Are you okay with where I put the endpoints?

I didn't make the email submission async in the mail service since an update to the status of the email submission needs to be saved to the database.

The email template is not very good, but I wanted to get your feedback about what content needs to be in it before I made it nicer.

### CODE REVIEW

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java [hunk 1]
```diff
@@ -183,4 +189,60 @@ public TerminationEmailDTO getTerminationWarningEmail(@PathVariable Long company
     public void sendTerminationWarningEmail(@Valid @RequestBody TerminationEmailDTO terminationEmailDTO) throws MessagingException {
         this.mailService.sendEmail(terminationEmailDTO);
     }
+
+    @PostMapping("mails/termination-notification")
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java:193]:
```suggestion
    @PostMapping("mails/company-termination-notification")
```

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java [hunk 2]
```diff
@@ -183,4 +189,60 @@ public TerminationEmailDTO getTerminationWarningEmail(@PathVariable Long company
...
+    public void sendTerminationNotificationEmail(@Valid @RequestBody List<Long> companyIds) throws MessagingException {
+        List<CompanyDTO> allCompanies =  companyService.findAll();
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java:195]: We don't have a service to fetch companies based on ids?

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java [hunk 3]
```diff
@@ -183,4 +189,60 @@ public TerminationEmailDTO getTerminationWarningEmail(@PathVariable Long company
...
+            companyVM.setCompanyType(company.getCompanyType());
+            companyVM.setLicenseType(company.getLicenseType());
+            companyVM.setLicenseModel(company.getLicenseModel());
+            companyVM.setLicenseStatus(company.getLicenseStatus());
+            companyVM.setBusinessContact(company.getBusinessContact());
+            companyVM.setLegalContact(company.getLegalContact());
+            companyVM.setCompanyDomains(company.getCompanyDomains());
+            companyVM.setAdditionalInfo(company.getAdditionalInfo());
+
+            companyService.updateCompany(companyVM);
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java:244]: What's the purpose here?

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/CompanyResource.java [hunk 4]
```diff
@@ -195,4 +201,35 @@ public ResponseEntity<Void> deleteCompany(@PathVariable Long id) {
...
+        log.debug("Request to get Company license is about to expire");
+        Instant now = Instant.now();
+        ArrayList<Long> companies = new ArrayList<>();
+        for (CompanyDTO company : companyService.findAll()) {
+           CompanyAdditionalInfoDTO additionalInfo = company.getAdditionalInfo();
+           if (additionalInfo != null) {
+               CompanyLicense license = additionalInfo.getLicense();
+               if (license != null) {
+                   CompanyTermination termination = license.getTermination();
+                   if (termination != null) {
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/web/rest/CompanyResource.java:216]: You can do Optional chaining to make it more readable. https://stackoverflow.com/a/35337260

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java (hunk 3 above)
Comment (jfkonecn) [reply src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java:244]: mark the email as sent

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java (hunk 2 above)
Comment (jfkonecn) [reply src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java:195]: As far as I can tell, [yes](https://github.com/oncokb/oncokb-public/blob/master/src/main/java/org/mskcc/cbio/oncokb/service/CompanyService.java).

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java (hunk 1 above)
Comment (jfkonecn) [reply src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java:193]: Should I change 188 too?
```java
@PostMapping("mails/termination-warning")
```

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java (hunk 2 above)
Comment (zhx828) [reply src/main/java/org/mskcc/cbio/oncokb/web/rest/MailsController.java:195]: I think we should add one.

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/CompanyResource.java (hunk 4 above)
Comment (jfkonecn) [reply src/main/java/org/mskcc/cbio/oncokb/web/rest/CompanyResource.java:216]: image

**Review by zhx828 — APPROVED** [review]
Thanks

**Pull request merged**